app/src/androidTest/java/org/kiwix/kiwixmobile/main/KiwixMainRobot.kt

```diff
@@ -0,0 +1,139 @@
+import org.kiwix.kiwixmobile.testutils.TestUtils.getResourceString
+import org.kiwix.kiwixmobile.utils.StandardActions.openDrawer
+
+fun kiwixMainRobo(func: KiwixMainRobot.() -> Unit) =
+  KiwixMainRobot().applyWithViewHierarchyPrinting(func)
+
+class KiwixMainRobot : BaseRobot() {
+
+  init {
+    isVisible(ViewId(R.id.navigation_container))
+  }
+
+  fun clickReaderOnBottomNav() {
+    clickOn(ViewId(R.id.readerFragment))
+  }
+
+  fun assertReaderScreenDisplayed() {
+    isVisible(Text(getResourceString(R.string.reader)))
+  }
+
+  fun clickLibraryOnBottomNav() {
+    clickOn(ViewId(R.id.libraryFragment))
+  }
+
+  fun library(func: LibraryRobot.() -> Unit) =
+    LibraryRobot().applyWithViewHierarchyPrinting(func)
+
+  inner class LibraryRobot : BaseRobot() {
+    init {
+      isVisible(Text(getResourceString(R.string.library)))
+    }
+
+    fun clickFileTransferIcon() {
+      clickOn(ViewId(R.id.get_zim_nearby_device))
+    }
+
+    fun assertFileTransferScreenDisplayed() {
+      isVisible(Text("Receive Files"))
+    }
+  }
+
+  fun clickDownloadOnBottomNav() {
+    clickOn(ViewId(R.id.downloadsFragment))
+  }
+
+  fun download(func: DownloadRobot.() -> Unit) =
+    DownloadRobot().applyWithViewHierarchyPrinting(func)
+
+  inner class DownloadRobot : BaseRobot() {
+    init {
+      isVisible(Text(getResourceString(R.string.download)))
+    }
+
+    fun clickOnGlobeIcon() {
+      clickOn(ViewId(R.id.select_language))
+    }
+
+    fun assertLanguagesScreenDisplayed() {
+      isVisible(Text(getResourceString(R.string.select_languages)))
+    }
+  }
+
+  fun clickBookmarksOnNavDrawer() {
+    clickOn(Text(getResourceString(R.string.bookmarks)))
+  }
+
+  fun assertBookmarksScreenDisplayed() {
+    isVisible(Text(getResourceString(R.string.bookmarks_from_current_book)))
+  }
+
+  fun clickHistoryOnSideNav() {
+    clickOn(Text(getResourceString(R.string.history)))
+  }
+
+  fun assertHistoryScreenDisplayed() {
+    isVisible(Text(getResourceString(R.string.history_from_current_book)))
+  }
+
+  fun clickHostBooksOnSideNav() {
+    clickOn(Text(getResourceString(R.string.menu_host_books)))
+  }
+
+  fun assertHostBooksScreenDisplayed() {
+    isVisible(Text(getResourceString(R.string.menu_host_books)))
+  }
+
+  fun clickSettingsOnSideNav() {
+    clickOn(Text(getResourceString(R.string.menu_settings)))
+  }
+
+  fun assertSettingsScreenDisplayed() {
+    isVisible(Text(getResourceString(R.string.menu_settings)))
+  }
+
+  fun clickHelpOnSideNav() {
+    clickOn(Text(getResourceString(R.string.menu_help)))
+  }
+
+  fun assertHelpScreenDisplayed() {
+    isVisible(Text(getResourceString(R.string.menu_help)))
+  }
+
+  fun clickSupportKiwixOnSideNav() {
+    clickOn(Text(getResourceString(R.string.menu_support_kiwix)))
+  }
+
+  fun assertExternalLinkDialogDisplayed() {
+    isVisible(Text(getResourceString(R.string.external_link_popup_dialog_title)))
+  }
+
+  fun navigate() {
+    pressBack()
+    openDrawer()
+  }
```

Comment:
I extracted it out in a function as it was being used at multiple places. Another name for it I have in mind is `recoverPreviousState()`, does any sound reasonable/logical enough?

Reply:
`navigate` can mean anything so yes we should rename it, if we follow the advice in my previous comment I think this method will disappear as we move the back nav  responsibility into the click receivers and this would just become `openDrawer` which doesn't need a method to wrap it